jib-core/src/main/java/com/google/cloud/tools/jib/registry/RegistryClient.java

```diff
@@ -367,6 +367,31 @@ private Authorization refreshBearerAuth(@Nullable String wwwAuthenticate)
     return Verify.verifyNotNull(initialBearerAuthenticator.get()).authenticatePush(credential);
   }
 
+  /**
+   * Check if an image is on the registry.
+   *
+   * @param <T> child type of ManifestTemplate
+   * @param manifestTemplateClass the specific version of manifest template to check
+   * @param imageDigest the image digest to check for
+   * @return the image's {@link ManifestAndDigest} if the image exists on the registry, or {@link
+   *     Optional#empty()} if it doesn't
+   * @throws IOException if communicating with the endpoint fails
+   * @throws RegistryException if communicating with the endpoint fails
+   */
+  public <T extends ManifestTemplate> Optional<ManifestAndDigest<T>> checkImage(
+      String imageDigest, Class<T> manifestTemplateClass) throws IOException, RegistryException {
```

Comment:
I didn't find any usage of passing `manifestTemplateClass` in this PR. Maybe just
```
  public Optional<ManifestAndDigest<ManifestTemplate>> checkImage(String imageDigest)
```
?